spec/_plugins/author_generator_spec.rb

```diff
@@ -26,12 +26,110 @@ def config
       'baseurl' => 'baseurl' 
     }
   end  
+end
+
+describe "Generate Author" do
+
+  it 'fails to generate if no author index' do
+    site = double 
+    allow(site).to receive(:config).and_return({})
+    allow(site).to receive(:layouts).and_return({'no_author_index' => nil})
+    genAuthor = Jekyll::GenerateAuthor.new   
+    expect{genAuthor.generate(site)}.to raise_exception(UncaughtThrowError, /uncaught throw "No 'author_index' layout found."/)
+  end  
+
+  it 'GenerateAuthor calls write_author_indexes with no posts and no collections' do
```

Comment:
I can't see an expectation of a call  to write_author_indexes as the name suggests, but it's probably my lack of knowledge of ruby